test/integration/workload/test.sh

```diff
@@ -30,51 +30,64 @@ workload_test() { # WORKLOAD_KIND
     echo "::endgroup::"
     single_test "w_${KIND}_${APIVERSION}_${TAG}" "Testing ${KIND} using ${APIVERSION} and ${TAG} image..." "workload" "${KIND}" "default" " created" "null"
 
-    if [[ "${GITHUB_BASE_REF:-"no-github-base-ref-set"}" == "master" || "${GITHUB_BASE_REF:-"no-github-base-ref-set"}" == "no-github-base-ref-set" ]]; then
+    if [[ "${GITHUB_BASE_REF:-"no-github-base-ref-set"}" == "master" || "${GITHUB_BASE_REF:-"no-github-base-ref-set"}" == "no-github-base-ref-set" || true ]]; then
+        wait_for_resource "${KIND}"
+        
         # Check that the workload object is actually ready, see #516
-        echo -n "Checking readiness of deployed resources..."
-        if [[ "${KIND}" == "StatefulSet" ]]; then
-            sleep 30 # StatefulSet provisions a PVC, which needs more time. A lot more sometimes...
-        fi
-        if [[ "${CI-}" == "true" ]]; then
-            SLEEP_TIME=5
+        echo -n "Checking if ${KIND} is ready..."
+        if check_resource_ready "${KIND}"; then
+            echo -e ${SUCCESS}
         else
-            SLEEP_TIME=20
+            echo -e ${FAILED}
+            echo "There are ${NUMBER_UNREADY} ${KIND} objects that aren't in a ready state:"
+            echo "${STATUSES}"
+            kubectl describe ${KIND} # Get us some debug information
+            EXIT="1"
         fi
-        sleep ${SLEEP_TIME}
+    fi
+}
 
-        # Output of different objects differs considerably, in particular in JSON representation
-        # To have less to differentiate, we parse the visual representation
-        if [[ "${KIND}" == "Pod" || "${KIND}" == "Deployment" || "${KIND}" == "StatefulSet" ]]; then
+check_resource_ready() { # RESOURCE_NAME
```

Comment:
Isn't it kind instead of name?
Could also do `KIND=${1}` and keep the better naming in vars later

Reply:
It still says name instead of kind here
```suggestion
check_resource_ready() { # WORKLOAD_KIND
```